### PR TITLE
Update integrations.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Added documentation process to this doc for better context.

This does need some work especially in terms of defining apps, plugins, and integrations but it's a start.

### CODE REVIEW

File: operations/research-and-development/engineering/integrations.md [hunk 1]
```diff
@@ -104,3 +122,17 @@ For both options, please ensure that doc updates are either provided by you or r
 The plugin documentation will soon be moved to developers.mattermost.com and will follow the same process as above. Until then, if you have a documentation update for your plugin, please work with @justinegeffen to ensure changes made in time and aligned with release.
 
 This is the same process used for the webapp and server repos in terms of documentation work.
+
+### Documentation for plugins with multiple versions
+
+Different plugins have different branching strategies, depending on the versioning pattern they use. The master branch is the source of truth of the docs for the "current release" of a plugin, and archive "previous release" as a (version)-docs branch when docs need to change.
```
Comment (hanzei) [inline operations/research-and-development/engineering/integrations.md]: Do they really have different branching strategies?

File: operations/research-and-development/engineering/integrations.md [hunk 2]
```diff
@@ -89,6 +89,24 @@ Seeing an intermittent connection issue with Zoom on http://dkh-local.ngrok.io.
 - Staff members add the issues there are working on for the current sprint to https://github.com/orgs/mattermost/projects/1.
     - Putting an issue is always preferred over putting a PR on the board.
 
+## Documentation process
+
+**1% draft**
+
+This guide outlines the processes and workflows for requesting, updating, managing, and maintaining documentation for Mattermost-managed and community-managed apps, plugins, and integrations. This includes the branching strategy for the Gitbook repos, how to set up a new Gitbook, and the process for aligning documentation with releases.
+
+Integrations comprise Mattermost-managed plugins and community-managed integrations.
```
Comment (hanzei) [inline operations/research-and-development/engineering/integrations.md]: I'm stumbling on the term "integrations". What does it refer to in this context?

Comment (hanzei) [inline operations/research-and-development/engineering/integrations.md]: I'm stumbling on the term "integrations". What does it refer to in this context?

File: operations/research-and-development/engineering/integrations.md [hunk 3]
```diff
@@ -89,6 +89,24 @@ Seeing an intermittent connection issue with Zoom on http://dkh-local.ngrok.io.
...
+
+- Mattermost-managed integrations are maintained by Mattermost staff and the documentation is written and maintained by the Mattermost Technical Writer.
+- Community-managed integrations are maintained by their creator, which includes documentation maintenance.
+
+Mattermost integrations documentation currently spans:
+
+- docs.mattermost.com: Outline of how integrations work with Mattermost and a list of integrations available.
+- Apps Marketplace: Content for specific apps, which is hosted in developers.mattermost.com.
```
Comment (hanzei) [inline operations/research-and-development/engineering/integrations.md]: I don't get what you a referring to. The Marketplace doesn't has any docs in it. Are you referring to the Apps Framework?

File: operations/research-and-development/engineering/integrations.md [hunk 4]
```diff
@@ -89,6 +89,24 @@ Seeing an intermittent connection issue with Zoom on http://dkh-local.ngrok.io.
...
+- Gitbook: Configuration and User documentation for plugins, as well as developer content.
+- developers.mattermost.com: Introduction to the contribution process, contribution workflow, testing and validation information.
```
Comment (hanzei) [inline operations/research-and-development/engineering/integrations.md]: Should plugin and app docs be mentioned here?

**Review by hanzei — CHANGES_REQUESTED** [review]

File: operations/research-and-development/engineering/integrations.md (hunk 1 above)
Comment (justinegeffen) [inline operations/research-and-development/engineering/integrations.md]:
```suggestion
If a plugin has documentation for multiple versions the master branch is the source of truth of the docs for the current release version of a plugin. Previous releases are archived as a (version)-docs branch when docs need to change.
```

File: operations/research-and-development/engineering/integrations.md [hunk 5]
```diff
@@ -104,3 +122,17 @@ For both options, please ensure that doc updates are either provided by you or r
 The plugin documentation will soon be moved to developers.mattermost.com and will follow the same process as above. Until then, if you have a documentation update for your plugin, please work with @justinegeffen to ensure changes made in time and aligned with release.
 
 This is the same process used for the webapp and server repos in terms of documentation work.
+
+### Documentation for plugins with multiple versions
+
+Different plugins have different branching strategies, depending on the versioning pattern they use. The master branch is the source of truth of the docs for the "current release" of a plugin, and archive "previous release" as a (version)-docs branch when docs need to change.
+
+Every time we release a new minor or major version plugin we cut a new branch with docs and keep the branch forever so customers can still refer back to it.
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/integrations.md:126]:
```suggestion
Every time we release a new minor or major version plugin we cut a new branch with docs and keep the branch forever so customers can still refer back to it.

Plugins that don't have documentation for multiple versions can have their content updated in master and don't follow this process.
```

File: operations/research-and-development/engineering/integrations.md [hunk 6]
```diff
@@ -104,3 +122,17 @@ For both options, please ensure that doc updates are either provided by you or r
...
+
+### Documentation for plugins with multiple versions
+
+If a plugin has documentation for multiple versions the master branch is the source of truth of the docs for the current release version of a plugin. Previous releases are archived as a (version)-docs branch when docs need to change.
+
+Every time we release a new minor or major version plugin we cut a new branch with docs and keep the branch forever so customers can still refer back to it.
+
+### Working in Gitbook
+
+We use Gitbook to render our documentation.
```
Comment (justinegeffen) [inline operations/research-and-development/engineering/integrations.md]:
```suggestion
We use Gitbook to render our documentation. A migration process from Gitbook to the Developers website is scheduled to start soon, at which point all plugin documentation will live in that repo. 

Details of the PR process for updating docs once this migration is complete will follow.
```

File: operations/research-and-development/engineering/integrations.md (hunk 4 above)
Comment (justinegeffen) [inline operations/research-and-development/engineering/integrations.md]:
```suggestion
- developers.mattermost.com: Apps documentation, plugin documentation, introduction to the contribution process, contribution workflow, testing, and validation information.
```

File: operations/research-and-development/engineering/integrations.md (hunk 3 above)
Comment (justinegeffen) [reply operations/research-and-development/engineering/integrations.md]: No, this was referring to future plans where content for apps would be displayed in the Marketplace and live in the developer-documentation repo. It's not applicable at this stage though.

File: operations/research-and-development/engineering/integrations.md [hunk 7]
```diff
@@ -89,18 +89,50 @@ Seeing an intermittent connection issue with Zoom on http://dkh-local.ngrok.io.
...
 
+## Documentation process
+
+**1% draft**
+
+This guide outlines the processes and workflows for requesting, updating, managing, and maintaining documentation for Mattermost-managed and community-managed apps, plugins, and integrations. This includes the branching strategy for the Gitbook repos, how to set up a new Gitbook, and the process for aligning documentation with releases.
+
+Mattermost integrations documentation currently spans:
+
+- docs.mattermost.com: Outline of how integrations work with Mattermost and a list of integrations available.
```
Comment (hanzei) [inline operations/research-and-development/engineering/integrations.md]: Should we link to https://docs.mattermost.com/guides/integration.html instead?

**Review by hanzei — DISMISSED** [review]
LGTM

File: operations/research-and-development/engineering/integrations.md (hunk 7 above)
Comment (justinegeffen) [inline operations/research-and-development/engineering/integrations.md]:
```suggestion
- [docs.mattermost.com](https://docs.mattermost.com/guides/integration.html): Outline of how integrations work with Mattermost and a list of integrations available.
```

**Review by hanzei — APPROVED** [review]

**Pull request merged**